libyul/AsmAnalysis.cpp

```diff
@@ -425,6 +425,19 @@ size_t AsmAnalyzer::operator()(FunctionCall const& _funCall)
 							"The \"verbatim_*\" builtins cannot be used with empty bytecode."
 						);
 				}
+				else if (functionName == "eofcreate" || functionName == "returncontract")
+				{
+					auto const& argumentAsLiteral = std::get<Literal>(arg);
+					// FIXME: m_dataNames contains root object name which means we can call EOF creation function with
+					// root container which is wrong. Is should be possible to call these functions with direct
+					// subcontainers only.
```

Comment:
Naming subcontainers with dots is just a convention and it's not required by yul right? If so we cannot rely in this assumption. I will pass current object name to the analyzer.

Reply:
> Naming subcontainers with dots is just a convention and it's not required by yul right? 

Not really. We actually do assume that values passed to builtins like `datasize()` or `dataoffset()` are dot-separated paths and internally we split them on those dots: https://github.com/ethereum/solidity/blob/b16d811921c6d3b2a2a9eba28b54a8af13bb28fa/libyul/Object.cpp#L148

I actually assumed that dots are not allowed in object names but apparently that's not the case. I think it's a bug though, because such names are not even usable with builtins. E.g.  `datasize()` will give you the `Unknown data object` error if you try to use them. We're going to have to fix that, but that may be breaking. But since EOF is a new thing, we can actually easily change it just for it - just add an extra check in `AsmAnalysis` that enforces no dots in object names on EOF (you could add that to #15536).

With that you can just check for dots here.

Reply:
#15540